### PR TITLE
Fixes 212: Add sort/search to rpm List

### PR DESCRIPTION
**Pull request by Andrewgdewar**
Adds sort/search functionality to the rpm list endpoint. 

This PR is paired with the front-end PR found [here.](https://github.com/content-services/content-sources-frontend/pull/46)

Still needs: 
- [x] Tests

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-212

File: pkg/api/repository_rpm.go [hunk 1]
```diff
@@ -17,6 +17,12 @@ type RepositoryRpmCollectionResponse struct {
 	Links Links            `json:"links"` // Links to other pages of results
 }
 
+type RepositoryRpmRequest struct {
+	UUID   string `param:"uuid"`    // Identifier of the rpm
```
Comment (jlsherrill) [inline pkg/api/repository_rpm.go:21]: This is the repository uuid right?

File: pkg/handler/repository_rpm.go [hunk 2]
```diff
@@ -74,16 +76,17 @@ func (rh *RepositoryRpmHandler) searchRpmPreprocessInput(input *api.SearchRpmReq
 // @Router       /repositories/{uuid}/rpms [get]
```
Comment (jlsherrill) [inline pkg/handler/repository_rpm.go:76]: need to add @Param options here similar to the search on repository list

Comment (Andrewgdewar) [reply pkg/handler/repository_rpm.go:76]: roger! 🖖

File: pkg/dao/rpm.go [hunk 3]
```diff
@@ -417,3 +429,47 @@ func FilteredConvert(yumPkgs []yum.Package, excludeChecksums []string) []models.
...
+		if ok {
+			// Concatenate (e.g. "url desc," + "name" + " asc")
+			sqlOrderBy = sqlOrderBy + sortField + ascOrDesc
+
+			// Add a comma if this isn't the last item in the "sortByArray".
+			if i+1 < lengthOfSortByParams {
+				sqlOrderBy = sqlOrderBy + ", "
+			}
+		}
+	}
```
Comment (jlsherrill) [inline pkg/dao/rpm.go:468]: can we pull most of this to a helper function that could also be used by  dao/repository_config.go?

Comment (Andrewgdewar) [reply pkg/dao/rpm.go:468]: I was thinking this too, but where would that helper function live?

File: pkg/handler/repository_rpm.go (hunk 2 above)
Comment (Andrewgdewar) [reply pkg/handler/repository_rpm.go:76]: add this here and to repository list... where I may have neglected that in the past. 🙄

File: pkg/dao/rpm.go (hunk 3 above)
Comment (rverdile) [reply pkg/dao/rpm.go:468]: I don't think there's a precedent for a function shared between two dao files (except in tests). Maybe a `dao/helpers.go` file?

File: pkg/dao/helpers.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package dao
+
+import "strings"
+
+func convertSortByToSQL(SortBy string, SortMap map[string]string) string {
```
Comment (rverdile) [inline pkg/dao/helpers.go:5]: you could add a test just for this function

Comment (Andrewgdewar) [reply pkg/dao/helpers.go:5]: Rog!

File: pkg/handler/repository_rpm.go [hunk 5]
```diff
@@ -70,20 +72,25 @@ func (rh *RepositoryRpmHandler) searchRpmPreprocessInput(input *api.SearchRpmReq
 // @Accept       json
 // @Produce      json
 // @Param		 uuid	path string true "Identifier of the Repository"
+// @Param		 limit query int false "Limit the number of items returned"
+// @Param		 offset query int false "Offset into the list of results to return in the response"
+// @Param		 search query string false "Search term for name and url."
```
Comment (jlsherrill) [inline pkg/handler/repository_rpm.go:77]:
```suggestion
// @Param		 search query string false "Search term for name."
```

File: pkg/dao/rpm.go (hunk 3 above)
Comment (jlsherrill) [reply pkg/dao/rpm.go]: works4me!

File: pkg/dao/helpers_test.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+		"name":          "name",
+		"url":           "url",
+		"package_count": "banana",
+		"status":        "status",
+	}
+
+	asc := ":asc"
+	desc := ":desc"
+
+	result := convertSortByToSQL("name", sortMap)
```
Comment (rverdile) [inline pkg/dao/helpers_test.go:20]: looking at the function, am I right that it supports multiple sort parameters i.e. "name:asc,version:desc"? If so I think there should be a test for that

Comment (rverdile) [reply pkg/dao/helpers_test.go:20]: could also test passing it an empty string

Comment (Andrewgdewar) [reply pkg/dao/helpers_test.go:20]: That gave me a good idea about trimming whitespace..

File: pkg/dao/helpers_test.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+	result = convertSortByToSQL("url"+asc, sortMap)
+	assert.Equal(t, "url asc", result)
+
+	result = convertSortByToSQL("package_count", sortMap)
+	assert.Equal(t, "banana asc", result)
+
+	result = convertSortByToSQL("status"+desc, sortMap)
+	assert.Equal(t, "status desc", result)
+
+	result = convertSortByToSQL(" status , name:desc", sortMap)
```
Comment (Andrewgdewar) [inline pkg/dao/helpers_test.go:35]: This tests when multiple sortable items are listed, and also tests the new trimming of whitespace.

**Review by rverdile — COMMENTED** [review]
Testing changes lgtm. I tested this with the frontend and it seems to be working. I think that's all from me.

**Review by jlsherrill — APPROVED** [review]
ack from me, needs a rebase though
